src/Common/Core/Impl/OS/ProcessExtensions.cs

```diff
@@ -0,0 +1,14 @@
+﻿using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Common.Core.Tasks;
+
+namespace Microsoft.Common.Core.OS {
+    public static class ProcessExtensions {
+        public static Task WaitForExitAsync(this IProcess process, int milliseconds, CancellationToken cancellationToken = default(CancellationToken)) {
+            var tcs = new TaskCompletionSource<int>();
+            process.Exited += (o, e) => tcs.TrySetResult(0);
```

Comment:
Got it. Then this is fine.

---
In reply to: [133773230](https://github.com/Microsoft/RTVS/pull/3867#discussion_r133773230) [](ancestors = 133773230)